src/Containers/PaletteForm/PaletteForm.js

```diff
@@ -1,17 +1,84 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { postPalette } from '../../Thunks/postPalette';
+import { fetchProjectPalettes } from '../../Thunks/fetchProjectPalettes';
 
 export class PaletteForm extends Component{
+  constructor() {
+    super();
+    this.state = {
+      name: '',
+      projectId: ''
+    };
+  };
+
+  createPalette = () => {
+    const { name } = this.state;
+    const {currentPalette} = this.props
+    let paletteToAdd = currentPalette.reduce((acc, color, index) => {
+      acc[`color${index+1}`] = color.hex
+      return acc
+    }, {})
+    paletteToAdd = {...paletteToAdd, name}
+    return paletteToAdd
+  }
+
+  handleChange = (event) => {
+    const { value } = event.target;
+    this.setState({ name: value });
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const palette = this.createPalette();
+    const  projectId  = parseInt(this.state.projectId)
+    this.props.fetchProjectPalettes(projectId);
```

Comment:
Good call fetching the palettes for the project, in case that project's palettes haven't been stored in redux yet